src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/ActivityMappingResourceIntTest.kt

```diff
@@ -252,6 +253,196 @@ class ActivityMappingResourceIntTest : IntegrationTestBase() {
     }
   }
 
+  @DisplayName("PUT /mapping/activities")
+  @Nested
+  inner class UpdateMappingTest {
+
+    private fun createUpdateRequest(
+      activityMapping: Pair<Long, Long> = activityScheduleId to nomisCourseActivityId,
+      scheduleMappings: List<Pair<Long, Long>> = listOf(activityScheduledInstanceId to nomisCourseScheduleId),
+    ): String {
+      val mappingsJson = scheduleMappings.joinToString { mapping ->
+        """
+          {
+            "scheduledInstanceId": ${mapping.first},
+            "nomisCourseScheduleId":${mapping.second},
+            "mappingType": "ACTIVITY_UPDATED"
+          }
+        """.trimIndent()
+      }
+
+      return """
+          {
+            "activityScheduleId"    : ${activityMapping.first},
+            "nomisCourseActivityId" : ${activityMapping.second},
+            "mappingType"           : "ACTIVITY_UPDATED",
+            "scheduledInstanceMappings" : [$mappingsJson]
+          }
+      """.trimIndent()
+    }
+
+    @BeforeEach
+    fun setUp() {
+      runBlocking {
```

Comment:
I think for tests you're better of using `runTest` - https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/run-test.html